src/System.Text.Json/tests/JsonTestHelper.cs

```diff
@@ -577,5 +577,24 @@ public static string GetCompactString(string jsonString)
                 }
             }
         }
+
+        public delegate void AssertThrowsActionUt8fJsonReader(Utf8JsonReader json);
+
+        // Cannot use standard Assert.Throws() when testing Utf8JsonReader - ref structs and closures don't get along.
+        public static void AssertThrows<E>(Utf8JsonReader json, AssertThrowsActionUt8fJsonReader action) where E : Exception
+        {
+            try
+            {
+                action(json);
+                Assert.False(true, "Expected exception: " + typeof(E).GetType());
+            }
+            catch (E)
+            {
+            }
```

Comment:
You want to add `Assert.IsType<E>(e)` here; otherwise this is actually `AssertThrowsAny` (it allows derived types to pass)